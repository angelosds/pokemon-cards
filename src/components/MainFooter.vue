<template>
  <footer class="main-footer">
    <span class="main-footer__text">
      <slot />
    </span>
    <button class="main-footer__button" type="button" @click="onClick">
      {{ actionText }}
    </button>
  </footer>
</template>

<script lang="ts">
import { Component, Emit, Prop, Vue } from "vue-property-decorator";

@Component
export default class MainFooter extends Vue {
  @Prop() private readonly actionText: string;

  @Emit()
  onClick(event: Event) {
    return event;
  }
}
</script>

<style lang="scss" scoped>
.main-footer {
  align-items: center;
  display: flex;
  height: 70px;
  justify-content: space-between;
  margin: 0 auto;
  max-width: 1024px;
  padding: 0 24px;

  &__button {
    background-color: transparent;
    border: 0;
    color: $text-color;
    font-size: 1.6rem;
    text-decoration: underline;
    cursor: pointer;
    font-family: $font-family-roboto;
    outline: 0;
  }
}
</style>
