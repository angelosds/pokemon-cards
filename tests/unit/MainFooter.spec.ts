import { shallowMount } from "@vue/test-utils";

import MainFooter from "@/components/MainFooter.vue";

describe("MainFooter.vue", () => {
  const build = (propsData?: object, slots?: object) => {
    const wrapper = shallowMount(MainFooter, {
      propsData: { actionText: "button", ...propsData },
      slots: { ...slots },
      mocks: {
        $t: (text: string) => text
      }
    });

    const button = () => wrapper.find(".main-footer__button");
    const text = () => wrapper.find(".main-footer__text");

    return { button, text, wrapper };
  };

  it("renders the component", () => {
    const { wrapper } = build();

    expect(wrapper).toMatchSnapshot();
  });

  it("renders text in a defautl slot", () => {
    const footerText = "Copyright 2021";
    const { text } = build(undefined, { default: footerText });

    expect(text().text()).toMatch(footerText);
  });

  it("renders a different language in a button", () => {
    const languageText = "click to portuguese";
    const { button } = build({ actionText: languageText });

    expect(button().text()).toMatch(languageText);
  });

  it("emits an on-click event", () => {
    const { button, wrapper } = build();

    button().trigger("click");

    expect(wrapper.emitted("on-click")).toBeTruthy();
  });
});
