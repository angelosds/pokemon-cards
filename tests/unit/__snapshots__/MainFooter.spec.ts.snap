// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainFooter.vue renders the component 1`] = `
<footer class="main-footer"><span class="main-footer__text"></span> <button type="button" class="main-footer__button">
    button
  </button></footer>
`;
